include/xtensor/xstrided_view.hpp

```diff
@@ -588,20 +588,20 @@ namespace xt
     inline auto strided_view(E&& e, const xstrided_slice_vector& slices)
     {
         detail::strided_view_args<detail::no_adj_strides_policy> args;
-        args.fill_args(e.shape(), detail::get_strides(e), detail::get_offset(e), e.layout(), slices);
+        args.fill_args(e.shape(), detail::get_strides<XTENSOR_DEFAULT_TRAVERSAL>(e), detail::get_offset<XTENSOR_DEFAULT_TRAVERSAL>(e), e.layout(), slices);
         using view_type = xstrided_view<xclosure_t<E>, decltype(args.new_shape)>;
         return view_type(std::forward<E>(e), std::move(args.new_shape), std::move(args.new_strides), args.new_offset, args.new_layout);
     }
 
-    template <class E, class S>
+    template <layout_type L = XTENSOR_DEFAULT_TRAVERSAL, class E, class S>
     inline auto reshape_view(E&& e, S&& shape)
     {
         using shape_type = std::decay_t<S>;
         get_strides_t<shape_type> strides;
 
         xt::resize_container(strides, shape.size());
-        compute_strides(shape, XTENSOR_DEFAULT_LAYOUT, strides);
-        using view_type = xstrided_view<xclosure_t<E>, shape_type, std::decay_t<E>::static_layout, detail::flat_adaptor_getter<xclosure_t<E>>>;
+        compute_strides(shape, L, strides);
+        using view_type = xstrided_view<xclosure_t<E>, shape_type, L, detail::flat_adaptor_getter<xclosure_t<E>, L>>;
```

Comment:
This is what fixes the assignment: do not use `std::decay_t<E>::static_layout`.

Reply:
Use `XTENSOR_DEFAULT_LAYOUT` in both `compute_strides` and xstrided_view's second template argument.